### DISCUSSION TITLE
Self-sharing a photo from ente to ente app creates an unique copy

### DISCUSSION THREAD

**Discussion by UnderTheCarpet** [ente-io/ente#798] [Enhancements]
**Describe the bug**
To work around the issue ente-io/ente#789 a friend of mine tried to shared a photo with me that has been shared with them. Because they couldn’t find a way they used the share function of the OS from within the ente app to share the photo with the app itself.

It’s an interesting workaround but to their surprise this created an unique copy of the photo (which the duplicate finder cannot detect).

**To Reproduce**

1. select photo
2. press share button
3. choose ente app
4. choose album

**Expected behavior**
Self-sharing should probably not silently create a copy? From my perspective it makes sense it copies the file, so it is not really a technical bug. But I‘d argue it is a UX bug. Fixing ente-io/ente#789 could help preventing this workaround. However, to my friend it was a natural workflow so I think ente should present a dialogue asking for confirmation to really create a copy.

**Smartphone (please complete the following information):**
 - Device: Google Pixel 6 Pro (but should not be device specific)
 - OS: Android 13
 - App Version 0.8.56

**Comment by ua741** [comment]
> tried to shared a photo with me that has been shared with him.
> It’s an interesting workaround but to their surprise this created an unique copy of the photo (which the duplicate finder cannot detect).

The duplicate detector only tries to find duplicates in the photos that are owned by the user.
On their home screen, they might be seeing two instance of the same photo, but one of them must have an indicator that it's a shared picture.

**Comment by UnderTheCarpet** [comment]
Yes, one of the shared copies has a 'shared' indicator.

Can you confirm it indeed creates a copy and takes up twice the space of the original photo in the storage?

**Comment by ua741** [comment]
Yes, on the server, it will occupy twice the space since they are separate copies owned by two different users. However, for the calculation of an individual user's storage utilization, it won't be counted twice because the copies belong to two different users.

P.S.: Instead of sharing, your friend can also download the shared photo directly. If they have enabled backup for all files, that photo will be uploaded to their account.

**Comment by UnderTheCarpet** [comment]
In my case my the friend is part of my family account (I probably should have called it family member instead) so it is probably counted against my personal space shared with my family.

Users find unexpexted ways to solve their problem 😃
